### DISCUSSION TITLE
Typescript error: `createSignal` setter conflicts with node types

### DISCUSSION THREAD

**Discussion by davidjbng** [solidjs/solid#1273] [Q&A]
I am seeing a weird Typescript error when calling a signal's setter with an `HTMLElement` like this

```ts
const [element, setElement] = createSignal<HTMLElement>();
// Error: No overload matches this call.
setElement(document.createElement('div'));
```
The error only occurs after installing `@types/node` which seem to be conflicting with `createSignal`'s Setter function types.

See the repro for the full error:
https://stackblitz.com/edit/github-axjy92?file=src/index.tsx

I also found that vite causes `@node/types` to be included because `tsc --explainFiles` outputs this (see _tsc.log_ in the repro )
```
  Type library referenced via 'node' from file 'node_modules/.pnpm/vite@3.0.9/node_modules/vite/dist/node/index.d.ts' with packageId '@types/node/ts4.8/index.d.ts@18.8.5'
```

This might by solved by changing `tsconfig.json` so that `@types/node` is not included by Typescript in _/src_ but I haven't found out how.

**Comment by ryansolid** [comment]
Node types are probably there due to SSR capability. We have types likely related to stream rendering that likely use them. But this isn't an area I understand or know anything about.

**Reply by davidjbng** [reply]
To be clear I had to install `@node/types` intentionally to make this error occur.

In my original project they where included indirectly by installing `@playwright/test`.
